src/implementations/deploy/token-registry/token-registry.ts

```diff
@@ -1,38 +1,119 @@
+  });
+
   if (dryRun) {
-    const unsignedTx = factory.getDeployTransaction(registryName, registrySymbol, {});
-    const tx = await wallet.populateTransaction(unsignedTx);
-    await dryRunMode({
-      network,
-      gasPriceScale: gasPriceScale,
-      transaction: tx,
-    });
+    const estimatedGas: BigNumber = await factory.estimateGas.deploy(
+      deployContractAddress.tokenImplementation,
+      initParam
+    );
+    signale.info("Dry Run is unavailable for token reigstry deploy");
+    signale.info(`Estimated Gas Required: ${estimatedGas.toString()}`);
     process.exit(0);
   }
   const gasPrice = await wallet.provider.getGasPrice();
   signale.await(`Sending transaction to pool`);
-  const transaction = await factory.deploy(registryName, registrySymbol, { gasPrice: gasPrice.mul(gasPriceScale) });
-  trace(`Tx hash: ${transaction.deployTransaction.hash}`);
-  trace(`Block Number: ${transaction.deployTransaction.blockNumber}`);
-  signale.await(`Waiting for transaction ${transaction.deployTransaction.hash} to be mined`);
-  return transaction.deployTransaction.wait();
+
+  const transaction = await factory.deploy(deployContractAddress.tokenImplementation, initParam, {
+    gasPrice: gasPrice.mul(gasPriceScale),
+  });
+  trace(`Tx hash: ${transaction.hash}`);
+  trace(`Block Number: ${transaction.blockNumber}`);
+  signale.await(`Waiting for transaction ${transaction.hash} to be mined`);
+  const receipt = await transaction.wait();
+  const registryAddress = getEventFromReceipt<DeploymentEvent>(receipt, factory.interface.getEventTopic("Deployment"))
+    .args.deployed;
+  return { contractAddress: registryAddress };
+};
+
+interface Params {
+  name: string;
+  symbol: string;
+  deployer: string;
+}
+
+export const encodeInitParams = ({ name, symbol, deployer }: Params): string => {
+  return ethers.utils.defaultAbiCoder.encode(["string", "string", "address"], [name, symbol, deployer]);
+};
+
+const retrieveFactoryAddress = (chainId: number, factoryAddress: string | undefined): DeployContractAddress => {
+  const { contractAddress } = constants;
+
+  if (!chainId) {
+    throw new Error(`Invalid chain ID: ${chainId}`);
+  }
+
+  let titleEscrowFactory = factoryAddress;
+  const tokenImplementation = contractAddress.TokenImplementation[chainId];
+  const deployer = contractAddress.Deployer[chainId];
+
+  if (!tokenImplementation || !deployer) {
+    throw new Error(`ChainId ${chainId} currently is not supported. Use token-registry to deploy.`);
+  }
+
+  if (!titleEscrowFactory) {
+    titleEscrowFactory = contractAddress.TitleEscrowFactory[chainId];
+    if (!titleEscrowFactory) {
+      throw new Error(`ChainId ${chainId} currently is not supported. Supply a factory address.`);
+    }
+  }
+
+  return {
+    titleEscrowFactory: titleEscrowFactory,
+    tokenImplementation: tokenImplementation,
+    deployer: deployer,
+  } as DeployContractAddress;
+};
+
+export const getEventFromReceipt = <T extends TypedEvent<any>>(
```

Comment:
a test might be helpful here to read through whats to be expected and not from using this function